### PR TITLE
Support Redis Cache and Refactor Credential Cache Using Cache Annotations

### PR DESCRIPTION
**Pull request by baixinsui**
Fixed https://github.com/eclipse-xpanse/xpanse/issues/1810

**Enable redis cache:**
Cache of  flavor price in redis:
chrome_NTGW3LJEs3
Cache credentials in redis:
After created:
chrome_B9dsLdYf5Z
After updated:
powershell_R5V1LxoSxL
After expired：
chrome_Auba8v5GSE
Cache monitor metrics in redis:
image

**Disable redis cache using caffeine cache:**
image

### CODE REVIEW

File: modules/cache/src/main/java/org/eclipse/xpanse/modules/cache/CaffeineCacheConfig.java [hunk 1]
```diff
@@ -56,4 +69,32 @@ private Cache<Object, Object> getServiceFlavorPriceCache() {
...
+                    if (Objects.nonNull(key)) {
+                        log.info("Credential cache removed, key: {}, cause: {}", key, cause);
+                    }
+                })
+                .build();
+        return cache;
+    }
+
+    private Cache<Object, Object> getMonitorMetricsCache() {
+        Cache cache = Caffeine.newBuilder()
```
Comment (swaroopar) [inline modules/cache/src/main/java/org/eclipse/xpanse/modules/cache/CaffeineCacheConfig.java:89]: we can directly return the cache object here.

File: modules/cache/src/main/java/org/eclipse/xpanse/modules/cache/CaffeineCacheConfig.java [hunk 2]
```diff
@@ -56,4 +69,32 @@ private Cache<Object, Object> getServiceFlavorPriceCache() {
                 : DEFAULT_CACHE_EXPIRE_TIME_IN_MINUTES;
         return Caffeine.newBuilder().expireAfterWrite(duration, TimeUnit.MINUTES).build();
     }
+
+
+    private Cache<Object, Object> getCredentialsCache() {
+        Cache cache = Caffeine.newBuilder()
```
Comment (swaroopar) [inline modules/cache/src/main/java/org/eclipse/xpanse/modules/cache/CaffeineCacheConfig.java:75]: we can directly return the cache object here.

File: modules/cache/src/main/java/org/eclipse/xpanse/modules/cache/CaffeineCacheConfig.java [hunk 3]
```diff
@@ -56,4 +69,32 @@ private Cache<Object, Object> getServiceFlavorPriceCache() {
...
+                        log.info("Credential cache removed, key: {}, cause: {}", key, cause);
+                    }
+                })
+                .build();
+        return cache;
+    }
+
+    private Cache<Object, Object> getMonitorMetricsCache() {
+        Cache cache = Caffeine.newBuilder()
+                .expireAfterWrite(10, TimeUnit.MINUTES)
```
Comment (swaroopar) [inline modules/cache/src/main/java/org/eclipse/xpanse/modules/cache/CaffeineCacheConfig.java:90]: this expiry does not match the value in the below code. 
I think it is better to move this as a config value in properties file. 
infact, would be better to have this configuration in properties file.. similar to other caches 

https://github.com/eclipse-xpanse/xpanse/blob/abc2b1c8b62b13cdee5103ac230153818039b882/modules/cache/src/main/java/org/eclipse/xpanse/modules/cache/monitor/MonitorMetricsStore.java#L92

**Review by swaroopar — CHANGES_REQUESTED** [review]

File: modules/cache/src/main/java/org/eclipse/xpanse/modules/cache/CaffeineCacheConfig.java (hunk 2 above)
Comment (baixinsui) [reply modules/cache/src/main/java/org/eclipse/xpanse/modules/cache/CaffeineCacheConfig.java]: Updated.

File: modules/cache/src/main/java/org/eclipse/xpanse/modules/cache/CaffeineCacheConfig.java (hunk 1 above)
Comment (baixinsui) [reply modules/cache/src/main/java/org/eclipse/xpanse/modules/cache/CaffeineCacheConfig.java]: Updated.

File: modules/cache/src/main/java/org/eclipse/xpanse/modules/cache/CaffeineCacheConfig.java (hunk 3 above)
Comment (baixinsui) [reply modules/cache/src/main/java/org/eclipse/xpanse/modules/cache/CaffeineCacheConfig.java]: Updated as suggestion.

**Review by swaroopar — APPROVED** [review]
LGTM

**Pull request merged**